src/include/ikarus/finiteElements/mechanics/enhancedAssumedStrains.hh

```diff
@@ -348,41 +334,67 @@ namespace Ikarus {
       if (onlyDisplacementBase) return;
 
       auto strainFunction  = DisplacementBasedElement::getStrainFunction(par);
-      auto C               = DisplacementBasedElement::getMaterialTangentFunction(par);
-      auto geo             = localView().element().geometry();
       const auto& numNodes = DisplacementBasedElement::numberOfNodes;
-
       assert(((numNodes == 4 and Traits::mydim == 2) or (numNodes == 8 and Traits::mydim == 3))
              && "EAS only supported for Q1 or H1 elements");
 
       std::visit(
-          [&]<typename EAST>(const EAST& easfunction) {
+          [&]<typename EAST>(const EAST& easFunction) {
             constexpr int enhancedStrainSize = EAST::enhancedStrainSize;
             Eigen::Matrix<double, enhancedStrainSize, enhancedStrainSize> D;
-
             D.setZero();
```

Comment:
this setZero should be inside calculateDAndLMatrix since it belongs to the calculation you can put it to LMat.setZero